Handlebars example from the docs appears not to compile

I have [created a demo repo ](https://github.com/MrTact/ft-bug)which contains the complete implementation of the handlebars demo from the docs.

When I try to build this repo, I get a pretty gnarly type conversion error:
```
error[E0277]: expected a `std::ops::Fn<(&handlebars::render::Helper<'reg, 'rc>, &'reg handlebars::registry::Registry<'reg>, &'rc handlebars::context::Context, &mut handlebars::render::RenderContext<'reg, 'rc>, &mut dyn handlebars::output::Output)>` closure, found `fluent_templates::loader::FluentLoader<&fluent_templates::loader::static_loader::StaticLoader>`
  --> src/main.rs:15:42
   |
15 |     handlebars.register_helper("fluent", Box::new(FluentLoader::new(&*LOCALES)));
   |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected an `Fn<(&handlebars::render::Helper<'reg, 'rc>, &'reg handlebars::registry::Registry<'reg>, &'rc handlebars::context::Context, &mut handlebars::render::RenderContext<'reg, 'rc>, &mut dyn handlebars::output::Output)>` closure, found `fluent_templates::loader::FluentLoader<&fluent_templates::loader::static_loader::StaticLoader>`
   |
   = help: the trait `for<'r, 'reg, 'rc, 's, 't0> std::ops::Fn<(&'r handlebars::render::Helper<'reg, 'rc>, &'reg handlebars::registry::Registry<'reg>, &'rc handlebars::context::Context, &'s mut handlebars::render::RenderContext<'reg, 'rc>, &'t0 mut (dyn handlebars::output::Output + 't0))>` is not implemented for `fluent_templates::loader::FluentLoader<&fluent_templates::loader::static_loader::StaticLoader>`
   = note: required because of the requirements on the impl of `handlebars::helpers::HelperDef` for `fluent_templates::loader::FluentLoader<&fluent_templates::loader::static_loader::StaticLoader>`
   = note: required for the cast to the object type `dyn handlebars::helpers::HelperDef + std::marker::Send + std::marker::Sync`
```

I would have submitted a PR for this, but frankly it's a bit beyond my understanding at the present time.

Note that I tried to build this both with 1.44 and nightly, so I don't think it's a recent feature that the lib is relying on.

--- a/templates/src/lib.rs
+++ b/templates/src/lib.rs
@@ -170,6 +170,10 @@
 //! automatically convert argument keys from Tera's `snake_case` to the fluent's
 //! preferred `kebab-case` arguments.
 //!
+//! ```toml
+//!fluent-templates = { version = "*", features = ["tera"] }
+//!```
+//!
 //! ```rust
 //! use fluent_templates::{FluentLoader, static_loader};
 //!
@@ -203,6 +207,10 @@
 //! ### Handlebars
 //! In handlebars, `fluent-templates` will read the `lang` field in your
 //! [`handlebars::Context`] while rendering.
+//!
+//! ```toml
+//!fluent-templates = { version = "*", features = ["handlebars"] }
+//!```
 //!
 //! ```rust
 //! use fluent_templates::{FluentLoader, static_loader};

